server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -381,6 +379,106 @@ func (s *BuildBuddyServer) JoinGroup(ctx context.Context, req *grpb.JoinGroupReq
 	return &grpb.JoinGroupResponse{}, nil
 }
 
+func (s *BuildBuddyServer) GetApiKeys(ctx context.Context, req *akpb.GetApiKeysRequest) (*akpb.GetApiKeysResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	tableKeys, err := userDB.GetAPIKeys(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	protoKeys := make([]*akpb.ApiKey, 0, len(tableKeys))
+	for _, k := range tableKeys {
+		protoKeys = append(protoKeys, &akpb.ApiKey{
+			Id:    k.APIKeyID,
+			Value: k.Value,
+			Label: k.Label,
+		})
+	}
+	return &akpb.GetApiKeysResponse{
+		ApiKey: protoKeys,
+	}, nil
```

Comment:
This is fine like you have it, but you can also just append these directly to your response like:

rsp := &akpb.GetApiKeysResponse{
	ApiKey: make([]*akpb.ApiKey, 0, len(tableKeys)),
}
for ... {
  rsp.ApiKey = append(rsp.ApiKey, key{})
}

return rsp, nil

Reply:
Done